### PR TITLE
Addition of React and front-end build system

### PR DESCRIPTION
**Pull request by iheanyi**
Finally integrated with React and all that good stuff, with a front-end build system. Go crazy.

### CODE REVIEW

File: Vagrantfile [hunk 1]
```diff
@@ -5,5 +5,7 @@ Vagrant.configure(2) do |config|
   config.vm.box = 'hashicorp/precise64'
   config.vm.network :forwarded_port, guest: 8000, host: 4567
   config.vm.network :forwarded_port, guest: 80, host: 8080
+  config.vm.network :forwarded_port, guest: 3000, host: 3000
```
Comment (GRardB) [inline Vagrantfile]: What's this for?

File: Vagrantfile [hunk 2]
```diff
@@ -5,5 +5,7 @@ Vagrant.configure(2) do |config|
...
   config.vm.provision :shell, path: 'provision/bootstrap.sh'
+  #config.vm.synced_folder ".", "/vagrant", type: "rsync"
```
Comment (GRardB) [inline Vagrantfile]: bruh

File: assets/js/app.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+  constructor(props) {
+    super(props);
+  }
+
+  render() {
+  
+    return (
+      <h1>Sup Hello World!</h1> 
+      
+      )
```
Comment (GRardB) [inline assets/js/app.js]: Why do you hate semicolons? What even is this indentation?

File: dispiff/settings.py [hunk 4]
```diff
@@ -28,6 +28,8 @@
 ALLOWED_HOSTS = []
 
 
+
+
```
Comment (GRardB) [inline dispiff/settings.py]: Why.

File: dispiff/settings.py [hunk 5]
```diff
@@ -105,3 +108,21 @@
 
 STATIC_URL = '/static/'
 STATIC_ROOT = os.path.join(BASE_DIR, "static/")
+STATICFILES_DIRS = [
+    os.path.join(BASE_DIR, "assets/"),
+]
```
Comment (GRardB) [inline dispiff/settings.py]: What's the difference between this and `STATIC_ROOT`? Do we need both?

File: index.html [hunk 6]
```diff
@@ -0,0 +1,11 @@
+<!DOCTYPE html>
+<html>
+    <head>
+      <title>DisPiff Webpack</title>
+    </head>
+
+    <body>
+      <div class="react-app"></div>
+      <script src="./assets/bundles/bundle.js"></script>
```
Comment (GRardB) [inline index.html]: Why do you hate absolute paths?

File: webpack.config.js [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+    loaders: [
+    { test: /\.(js|jsx)?$/, exclude: /node_modules/, loaders: ['react-hot','babel-loader']}, // to transform JSX into JS
+    ],
+  },
+
+  resolve: {
+    modulesDirectories: ['node_modules', 'bower_components'],
+    extensions: ['', '.js', '.jsx']
+  },
+}
```
Comment (GRardB) [inline webpack.config.js]: Semicolon pls.

File: webpack.local.config.js [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+config.plugins = config.plugins.concat([
+  new webpack.HotModuleReplacementPlugin(),
+  new webpack.NoErrorsPlugin(),
+  new BundleTracker({filename: './webpack-stats.json'}),
+])
+
+// Add a loader for JSX files with react-hot enabled
+config.module.loaders.push(
+  { test: /\.js?$/, exclude: /node_modules/, loaders: ['react-hot', 'babel'] }
+)
```
Comment (GRardB) [inline webpack.local.config.js]: SEMICOLON PLS EVERYWHERE.

File: webpack.prod.config.js [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+    }
+  })
+])
+
+// Add a loader for JSX files
+config.module.loaders.push(
+  { test: /\.js?$/, exclude: /node_modules/, loader: 'babel' }
+)
+
+module.exports = config
```
Comment (GRardB) [inline webpack.prod.config.js]: pls

**Comment by iheanyi** [comment]
@GRardB - I don't care for semi-colons in Webpack, but okay, okay. u.u

File: dispiff/settings.py (hunk 5 above)
Comment (iheanyi) [inline dispiff/settings.py]: STATIC_ROOT is the root directory, but you can have more than on STATIC_DIR, so you need to push them to this array.

File: index.html (hunk 6 above)
Comment (iheanyi) [inline index.html]: @GRardB - Smfh, because it's a web browser?

File: Vagrantfile (hunk 1 above)
Comment (iheanyi) [inline Vagrantfile]: `npm run watch`

**Comment by iheanyi** [comment]
@reggie Any insight? Haha

File: dispiff/urls.py [hunk 10]
```diff
@@ -16,6 +16,9 @@
 from django.conf.urls import include, url
 from django.contrib import admin
 
+from django.views.generic import TemplateView
+
 urlpatterns = [
+    url(r'^$', TemplateView.as_view(template_name='index.html')),
```
Comment (GRardB) [inline dispiff/urls.py]: I don't know much about how Django routes work yet, but will this match `example.com/`(with a trailing slash)? Or would we need to change it to `r'^\/?$` or something?

Comment (iheanyi) [inline dispiff/urls.py]: @GRardB  - It auto resolves.

**Comment by GRardB** [comment]
Ship it.

**Pull request merged**